File: benchmarks/nsieve/nsieve.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

func nsieve(n int) {
	flags := make([]bool, n)
	count := 0
	for i := 2; i < n; i++ {
		if !flags[i] {
			count++
			for j := i << 1; j < n; j += i {
				flags[j] = true
			}
		}
	}
	fmt.Printf("Primes up to %8d %8d\n", n, count)
}

func main() {
	n := 4
	if len(os.Args) > 1 {
		n, _ = strconv.Atoi(os.Args[1])
	}
	for i := range 3 {
		nsieve(10000 << (n - i))
	}
}

File: benchmarks/nsieve/nsieve.rs
fn nsieve(n: usize) {
    let mut flags = vec![false; n];
    let mut count = 0;
    for i in 2..n {
        if !flags[i] {
            count += 1;
            let mut j = i << 1;
            while j < n {
                flags[j] = true;
                j += i;
            }
        }
    }
    println!("Primes up to {:8} {:8}", n, count);
}

fn main() {
    let n: usize = std::env::args()
        .nth(1)
        .and_then(|s| s.parse().ok())
        .unwrap_or(4);

    for i in 0..3 {
        nsieve(10000 << (n - i));
    }
}
